src/main/java/Duke.java

```diff
@@ -1,10 +1,259 @@
+    public static final String USAGE_MARK = COMMAND_MARK + " <task index>";
+    public static final String USAGE_UNMARK = COMMAND_UNMARK + " <task index>";
+
+    // Program logic stuff
+    private static final Scanner in = new Scanner(System.in);
+
+    public static final int MAX_TASKS = 100;
+    private static Task[] savedTasks = new Task[MAX_TASKS];
+    private static int numSavedTasks = 0;
+
+    private static boolean isExitRequested = false;
+
+    /**
+     * Main entry point of the program.
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printMessage(MESSAGE_WELCOME);
+        while (!isExitRequested) {
+            String[] input = getUserInput();
+            String response = parseCommand(input);
+            printMessage(response);
+        }
+    }
+
+    /**
+     * Prints the specified message, followed by a horizontal separator on the next line.
+     * @param message the message to be printed
+     */
+    private static void printMessage(String message) {
+        System.out.println(message);
+        System.out.println(HORIZONTAL_SEPARATOR);
+    }
+
+    /**
+     * Reads user input and separates the command being invoked from its arguments.
+     * @return a String array of the form {command, args}
+     */
+    private static String[] getUserInput() {
+        System.out.print(INPUT_PROMPT);
+        String userInput = in.nextLine();
+        System.out.println(HORIZONTAL_SEPARATOR);
+        String[] userInputTokenized = userInput.trim().split(" ", 2);
+        if (userInputTokenized.length == 2) {
+            return userInputTokenized;
+        }
+        return new String[] {userInputTokenized[0], ""};
+    }
+
+    /**
+     * Generates an error message for an invalid command.
+     * @param usageExample string showing the correct command format
+     * @return generated error message
+     */
+    private static String invalidCommandError(String command, String usageExample) {
+        return String.format(MESSAGE_INCORRECT_COMMAND_FORMAT, command, usageExample);
+    }
+
+    /**
+     * Attempts to add the specified task to the task list.
+     * @param t task to be added
+     * @param successMessage message to be displayed on success
+     */
+    private static String addTask(Task t, String successMessage) {
+        if (numSavedTasks < MAX_TASKS) {
+            savedTasks[numSavedTasks] = t;
+            numSavedTasks++;
+            return String.format(successMessage, t);
+        }
+        return MESSAGE_TOO_MANY_TASKS;
+    }
+
+    /**
+     * Invokes the correct command function based on the command string inputted by the user.
+     */
```

Comment:
I noticed that some method comments include @params and @returns, but some only have the description of what the method is doing - consider having consistent structure for all of your comments.